### DISCUSSION TITLE
Accessing user working directory to read files from, for our extension to generate autocomplete..

### DISCUSSION THREAD

**Discussion by dnecklesportfolio** [microsoft/vscode-discussions#109] [Extension Development QnA] [answered]
Hello,
Reference to Line 86 https://github.com/oslabs-beta/SurfQL/blob/dev/src/extension.ts
My group has an autocomplete example where the graphql schema is currently in the same file as our extension. The autocomplete works  We want to read this same schema from an external file on the users computer, and eventually save it as a variable in our code, without them manually selecting it. Can someone point me towards any documentation which contains information on reading files from the user director?
I've tried these steps:  
I am open the extension project in vscode
Run extension. A new extension development host is window is opened.
I open a folder with an accompanying file (schema.graphqls) to be read by extension.  
But when I run the code below
 Error: cannot open file:///Users/dn/extension/dist/schema.graphqls. <- The issue is that the Point of reference is the extension not the users computer.
 
>  let uri = path.join(__dirname, 'schema.graphqls');
        console.log(__dirname)
        vscode.workspace.openTextDocument(uri).then((document) => {
          let text = document.getText();
          console.log(text)
        });

I know context.extensionPath gives the filepath of the directory containing the extension,
but I am looking to access the directory of the user's workspace outside of the extension directory.
Can someone point me towards any documentation which contains information on this?

**Comment by isidorn** [comment, marked as answer]
Try to use `vscode.workspace.fs`. API https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts#L11675
If you want to read things outside of the workspace - that is not really recommended. But you can use nodeJS to do that, for example nodeJS FS API.

**Comment by Tommycore** [comment]
Do you mean `vscode.workspace.workspaceFolders`? That would give you all folders the users current workspace contains. See https://code.visualstudio.com/api/references/vscode-api#workspace.workspaceFolders.

**Comment by dnecklesportfolio** [comment]
Awesome, thank you everyone, I think I should have what I'm looking for. I'll give it a shot.
